Formula/irace.rb

```diff
@@ -0,0 +1,39 @@
+class Irace < Formula
+  desc "Iterated Racing for Automatic Algorithm Configuration"
+  homepage "https://iridia.ulb.ac.be/irace"
+  url "https://cran.r-project.org/src/contrib/irace_3.4.1.tar.gz"
+  sha256 "7eea92ba42e6ba320fa8bdca3c53091ae42f26a0f097244f65e7e117f6d514b6"
+  license "GPL-2.0-or-later"
+
+  depends_on "r"
+
+  resource "R6" do
+    url "https://cran.r-project.org/src/contrib/R6_2.5.1.tar.gz"
+    sha256 "8d92bd29c2ed7bf15f2778618ffe4a95556193d21d8431a7f75e7e5fc102bf48"
+  end
+
+  def install
+    mkdir "#{buildpath}/r-lib"
+    resource("R6").stage do
+      system "R", "CMD", "INSTALL", ".", "--library=#{buildpath}/r-lib"
+    end
+    system "R", "CMD", "INSTALL", ".", "--library=#{buildpath}/r-lib"
+    prefix.install Dir["#{buildpath}/r-lib/"]
+    bin.install "#{prefix}/r-lib/irace/bin/irace"
+    bin.install "#{prefix}/r-lib/irace/bin/irace2pyimp"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace-mpi"
+    bin.install "#{prefix}/r-lib/irace/bin/parallel-irace-qsub"
+  end
+
+  def caveats
+    <<~EOS
+      You need to let R know the installation of the irace package by setting an environment variable (and add it to your ~/.profile, ~/.bashrc or ~/.zshrc for it to persist)
+        $ export R_LIBS="#{opt_prefix}/r-lib:${R_LIBS}"
+    EOS
+  end
+
+  test do
+    system "R_LIBS=#{opt_prefix}/r-lib:${R_LIBS} #{bin}/irace -v"
```

Comment:
```suggestion
    ENV["R_LIBS"] = opt_prefix}/"r-lib"
    system bin/"irace", "-v"
```

Additionally we need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.